### PR TITLE
Tests refactorings

### PR DESCRIPTION
**Pull request by HollowPrincess**
- [x] Add **README.md** with instructions for new tests addition  and other important information
- [x] Move  local `TESTS_ROOT_DIR` into the `seismiqb\tests'
- [x] Geometry tests fixes (fix seed, add new checks)
- [x] Add notebooks numbering
- [x] Add removal of the executed notebooks without failures
- [x] Change terminal log message (add an error cell number, change printing notebook name in logs on name with the  extension) 
- [x] Remove `TESTS_ROOT_DIR` dependence from notebooks
- [x] Refactor `CharismaMixin` tests code
- [x] Refactor `SeismicGeometry` tests code
- [x] Refactor `Horizon` tests code
- [x] Add tests template
- [x] Add more careful out filenames preparation and a file with names correspondence

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# A short instruction for new tests addition
+
+## Notebook preparation
+
+Before you add your test notebook to the running tests list, you need to make **simple preparations**:
+
+1. If your test saves any files, it is higly recommended to use relative paths and create a test's **own directory** for saving files.
```
Comment (SergeyTsimfer) [inline tests/README.md:7]: refrain from using `you / your`

`If the test notebook saves ...`

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+All files will be saved in the shared `TESTS_ROOT_DIR` directory (`'seismiqb\tests\test_root_dir_*'`), so a separate directory inside the `TESTS_ROOT_DIR` will prevent from mixing up files from different tests.
```
Comment (SergeyTsimfer) [inline tests/README.md:9]: saved to

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+2. All **changeable parameters** must be initialized on the first or on the second notebook cells. All manipulations with them must be done on cells number 3 or higher.
+
+This is caused because the `run_notebook_test.py` inserts a new cell with parameters initialization between cells number 2 and 3.
```
Comment (SergeyTsimfer) [inline tests/README.md:13]: `This is because`

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+So, the recommended notebook structure is:
+* **Cell 1**: necessary imports.
```
Comment (SergeyTsimfer) [inline tests/README.md:16]: `Cell #1`

File: tests/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+* **Cell 2**: parameters initialization.
+* **Other cells**: tests and additional manipulations.
```
Comment (SergeyTsimfer) [inline tests/README.md:18]: `Cells #3+`

**Review by SergeyTsimfer — APPROVED** [review]

**Review by cdtn — COMMENTED** [review]
Use `sys.path.insert` instead of `sys.path.append` in notebooks.

**Pull request merged**